classes/migration/upgrade/v3_4_0/I8933_EventLogLocalized.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I8933_EventLogLocalized.php
+ *
+ * Copyright (c) 2023 Simon Fraser University
+ * Copyright (c) 2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I8933_EventLogLocalized.php
+ *
+ * @brief Adds a column to the event_log_settings table to store localized data such as a file name and drops setting_type column.
+ * In the event_log table allows null values for userId.
+ * Fixes the issue with duplicate event types and renames conflicting setting names
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\log\event\PKPSubmissionEventLogEntry;
+use PKP\log\event\SubmissionFileEventLogEntry;
+use PKP\migration\Migration;
+
+abstract class I8933_EventLogLocalized extends Migration
+{
+    abstract protected function getContextTable(): string;
+
+    abstract protected function getContextIdColumn(): string;
+
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        Schema::table('event_log_settings', function (Blueprint $table) {
+            $table->string('locale', 14)->default('')->after('log_id');
+            $table->dropUnique('event_log_settings_unique');
+            $table->unique(['log_id', 'locale', 'setting_name'], 'event_log_settings_unique');
+            $table->string('setting_type', 6)->nullable()->change();
+        });
+
+        // Events can be triggered without a user, e.g., in schedule tasks
+        Schema::table('event_log', function (Blueprint $table) {
+            $table->dropForeign('event_log_user_id_foreign');
+            $table->dropIndex('event_log_user_id');
+            $table->bigInteger('user_id')->nullable()->change();
+            $table->foreign('user_id')->references('user_id')->on('users')->onDelete('cascade');
+            $table->index(['user_id'], 'event_log_user_id');
+            $table->boolean('is_translated')->change();
+        });
+
+        $this->fixConflictingSubmissionLogConstants();
+
+        // Rename ambiguous settings
+        $this->renameSettings();
+
+        // Localize existing submission file name entries
+        $sitePrimaryLocale = DB::table('site')->value('primary_locale');
+        DB::table('event_log_settings AS es')
+            ->join('event_log AS e', 'es.log_id', '=', 'e.log_id')
+            ->where('setting_name', 'filename')
+            ->whereIn('event_type', [
+                SubmissionFileEventLogEntry::SUBMISSION_LOG_FILE_UPLOAD,
```

Comment:
I would suggest hard-coding the constant values here, with e.g. `SubmissionFileEventLogEntry::SUBMISSION_LOG_FILE_UPLOAD` in an inline comment. This means grepping will still work but the constants will be able to evolve in future code without affecting the upgrade.